apps/consent-next.js/app/consent/page.tsx

```diff
@@ -0,0 +1,208 @@
+import { redirect } from "next/navigation";
+import React from "react";
+import { hydraClient } from "../hydra-config";
+import { oidcConformityMaybeFakeSession } from "../oidc-cert";
+
+interface ConsentProps {
+  consent_challenge: string;
+}
+
+const submitForm = async (form: FormData) => {
+  "use server";
+  const consent_challenge = form.get("consent_challenge");
+  const submitValue = form.get("submit");
+  const remember = form.get("remember") === "1";
+  const grantScope = form
+    .getAll("grant_scope")
+    .map((value) => value.toString());
+  // TODO add check from email code.
+
+  if (typeof consent_challenge === "string" && consent_challenge) {
```

Comment:
I think the if should testing the negative here and return an error if not valid

This would make the code cleaner as it would:
- return an error if condition is not true (we don't do anything here)
- avoid indenting the rest of the function. This makes it easier to reason about